### PR TITLE
pulumi: fix withPackages invocation

### PR DESCRIPTION
**Pull request by CameronNemo**
Re-add PATH handling, which was removed without explanation. (see #325527)
Also add an end quote to the LD_LIBRARY_PATH line.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mastermindaxe** [comment]
Builds successfully on `aarch64-darwin` 👌

File: pkgs/tools/admin/pulumi/default.nix [hunk 1]
```diff
@@ -131,7 +131,8 @@ buildGoModule rec {
       ''
         mkdir -p $out/bin
         makeWrapper ${pulumi}/bin/pulumi $out/bin/pulumi \
-          --set LD_LIBRARY_PATH "${stdenv.cc.cc.lib}/lib
+          --suffix PATH : ${lib.makeSearchPath "bin" (f pulumiPackages)} \
```
Comment (hraban) [inline pkgs/tools/admin/pulumi/default.nix:134]:
```suggestion
          --suffix PATH : ${lib.makeBinPath (f pulumiPackages)} \
```

**Review by hraban — APPROVED** [review]

**Comment by Mastermindaxe** [comment]
Maybe a dumb question, but when I use `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 331660"` it should produce a shell in which I can use the binary `pulumi` right? It builds, but I don't have a binary built in the shell. Am I remembering sth wrong or is there an error here?

**Comment by hraban** [comment]
I always use

```
nix shell github:nixos/nixpkgs/pull/331660/head#pulumi
```

**Comment by rupurt** [comment]
Thank you for this. It was driving me crazy trying to figure out why I wasn't getting the latest version.

**Comment by teto** [comment]
I was submitting my own PR when I noticed this one. I dunno about the PATH thing but since there is no answer on https://github.com/NixOS/nixpkgs/pull/325527#discussion_r1701063620 I think it's fine to merge.

I haven't tried yet but at some point we might want to do something like this as well to bypass pulumi's unbased warnings https://github.com/pulumi/pulumi/issues/14525#issuecomment-2322723229.

**Pull request merged**